examples/ucp_hello_world.c

```diff
@@ -128,11 +128,31 @@ static void recv_handler(void *request, ucs_status_t status,
            info->length);
 }
 
-static void ucx_wait(ucp_worker_h ucp_worker, struct ucx_context *context)
+static ucs_status_t ucx_wait(ucp_worker_h ucp_worker, struct ucx_context *context,
```

Comment:
using both `request` and `user_data`(`context`) looks confusing, especially when the same request object is passed twice.
Maybe rely on just `request` for now and check how to show usage of `user_data` later?

Reply:
done